bootstrap/handlers/secret/client/vaultclient_test.go

```diff
@@ -0,0 +1,381 @@
+			authToken:           "test-token",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         true,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with unrenewable token",
+			authToken:           "unrenewableToken",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       true,
+			expectError:         false,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with no TokenFile",
+			authToken:           "testToken1",
+			tokenFile:           "",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         false,
+			expectedErrorType:   nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			testSecretStoreInfo.TokenFile = test.tokenFile
+			testContainer.Update(
+				di.ServiceConstructorMap{
+					container.ConfigurationInterfaceName: func(get di.Get) interface{} {
+						return &configurationStruct{
+							secretStore: testSecretStoreInfo,
+						}
+					},
+				})
+			cfgHTTP := vault.SecretConfig{
+				Host:           host,
+				Port:           portNum,
+				Protocol:       "http",
+				Authentication: vault.AuthenticationInfo{AuthToken: test.authToken},
+			}
+
+			sclient := NewSecretVaultClient(bkgCtx, cfgHTTP, testContainer)
+			_, err := sclient.GetClient()
+
+			if test.expectedErrorType != nil && err == nil {
+				t.Errorf("Expected error %v but none was received", test.expectedErrorType)
+			}
+
+			if !test.expectError && err != nil {
+				t.Errorf("Unexpected error: %s", err.Error())
+			}
+
+			if test.expectError && test.expectedErrorType != nil && err != nil {
+				eet := reflect.TypeOf(test.expectedErrorType)
+				aet := reflect.TypeOf(err)
+				if !aet.AssignableTo(eet) {
+					t.Errorf("Expected error of type %v, but got an error of type %v", eet, aet)
+				}
+			}
+
+			tokenCallback := sclient.getDefaultTokenExpiredCallback(testSecretStoreInfo)
+			if test.expectedNilCallback && tokenCallback != nil {
+				t.Error("expected nil token expired callback func, but found not nil")
+			}
+			if !test.expectedNilCallback {
+				if tokenCallback == nil {
+					t.Error("expected some non-nil token expired callback func, but found nil")
+				} else {
+					repToken, retry := tokenCallback(test.authToken)
+
+					if repToken != test.expectedNewToken {
+						t.Errorf("expected a new token [%s] from callback but got [%s]", test.expectedNewToken, repToken)
+					}
+
+					if retry != test.expectedRetry {
+						t.Errorf("expected retry %v for a default token expired callback but got %v", test.expectedRetry, retry)
+					}
+
+					lc.Debug(fmt.Sprintf("repToken = %s, retry = %v", repToken, retry))
+				}
+			}
+		})
+	}
+	// wait for some time to allow renewToken to be run if any
+	time.Sleep(7 * time.Second)
+}
+
+func getMockTokenServer(tokenDataMap *sync.Map) *httptest.Server {
```

Comment:
This implementation should be moved into its own file.

Reply:
sure, good suggestion